tcms/templates/attachments/delete_link.html

```diff
@@ -1,4 +1,4 @@
 {% load i18n %}
 {% load static %}
 <link rel="stylesheet" href="{% static 'patternfly/dist/css/patternfly-additions.min.css' %}">
-{% if delete_url %}<a class="delete-attachment" href="{{ delete_url }}?next={{ next }}"><span class="pficon-error-circle-o"></span></a>{% endif %}
+{% if delete_url %}<a class="delete-attachment" onclick="promptAttachmentDeletion()" href="{{ delete_url }}?next={{ next }}"><span class="pficon-error-circle-o"></span></a>{% endif %}
```

Comment:
You have to `return function_call()` in `onclick` so that it can have any effect.

You should also make the confirmation string a parameter to the function and wrap it with a `{% trans %}` tag so that text is marked for translation.

Last, `utils.js` may not be loaded in every page where we see attachment files so the best thing to do here is to actually execute `confirm()` directly inside `onclick`.